tests/test_packages/test_connections/test_http/test_http.py

```diff
@@ -0,0 +1,347 @@
+            connection_id=cls.connection_id,
+            restricted_to_protocols=set([cls.protocol_id]),
+        )
+        cls.loop = asyncio.get_event_loop()
+        cls.http_connection.loop = cls.loop
+        value = cls.loop.run_until_complete(cls.http_connection.connect())
+        assert value is None
+        assert cls.http_connection.connection_status.is_connected
+        assert not cls.http_connection.channel.is_stopped
+
+    @pytest.mark.asyncio
+    async def test_get_404(self):
+        """Test send connection error."""
+
+        def request_response_cycle():
+            conn = http.client.HTTPConnection(self.host, self.port)
+            conn.request("GET", "/")
+            response = conn.getresponse()
+            return response.status, response.read()
+
+        async def client_thread():
+            executor = concurrent.futures.ThreadPoolExecutor(max_workers=3)
+            loop = asyncio.get_event_loop()
+            result = await loop.run_in_executor(executor, request_response_cycle)
+            return result
+
+        response_code, response_body = await client_thread()
+
+        assert response_code == 404 and response_body == b"Request Not Found"
+
+    @pytest.mark.asyncio
+    async def test_get_408(self):
+        """Test get 408."""
+
+        def request_response_cycle():
+            conn = http.client.HTTPConnection(self.host, self.port)
+            conn.request("GET", "/pets")
+            response = conn.getresponse()
+            return response.status, response.read()
+
+        async def client_thread():
+            executor = concurrent.futures.ThreadPoolExecutor(max_workers=3)
+            loop = asyncio.get_event_loop()
+            result = await loop.run_in_executor(executor, request_response_cycle)
+            return result
+
+        response_code, response_body = await client_thread()
+
+        assert response_code == 408 and response_body == b"Request Timeout"
+
+        envelope = await self.http_connection.receive()
+        assert (
+            envelope is not None
+            and envelope.to == self.address
+            and len(self.http_connection.channel.timed_out_request_ids) == 1
+        )
+
+    # @pytest.mark.asyncio
+    # async def test_get_202(self):
+    #     """Test send connection error."""
+
+    #     def request_response_cycle() -> Tuple[int, bytes]:
+    #         conn = http.client.HTTPConnection(self.host, self.port)
+    #         conn.request("GET", "/pets")
+    #         response = conn.getresponse()
+    #         return response.status, response.read()
+
+    #     async def client_thread() -> Tuple[int, bytes]:
+    #         executor = concurrent.futures.ThreadPoolExecutor(max_workers=3)
+    #         loop = asyncio.get_event_loop()
+    #         result = await loop.run_in_executor(executor, request_response_cycle)
+    #         return result
+
+    #     response_code, response_body = await client_thread()
```

Comment:
this line is not correct. That won't execute `agent_processing()` because it will `await` the result from the corouting.

What you can do is to "schedule" the task and run it concurrently with the `agent_processing`. thanks to the delay you wrote, the `client_thread` will send the HTTP connection before `agent_processing`reads it.

Reply:
see here one way to do it:
https://github.com/fetchai/agents-aea/pull/901/files#diff-9424a9106678c3c13e4c9501f9f458b7R243-R250

Reply:
Thanks :)